File: EgretProject/src/h5/BattleCache.ts
class BattleCache {
	public constructor() {
	}

	//用户信息
	userInfo: any;
    //攻击方名称
    attackName =  "";
    //被攻击方名称
	byAttackName = "";
    //战斗类型
    battleType = FightModuleEnum.FIGHT_TYPE_PVE_NORMAL;
    //攻击方战力
    attackCombatForce = 0;
    //被攻击方战力
    byAttackComobatForce = 0;
    //被攻击者Id
    byAttackerId = 0;
    //当前战斗场次
    currentBattleCount = 0;
    //最大战斗场次
    maxBattleCount = 0;
    //难度
    difficulty = 1;
    //阵型战斗对象
    attackerObjects: {[pos: number]: FightObject} = {};
    //被攻击者战斗对象队列
    byAttackerObjectsList: Array<{[pos: number]: FightObject}> = [];
    //战斗中奖励卡牌
    battleReward: any = null;
    //战斗中奖励道具
    battleRewardProp: any = null;
    //战斗中奖励装备
    battleRewardEquipment: any = null;
    //总伤害
    totalDamage = 0;
    userViceCaptainInfo: any = null;
    byUserViceCaptainInfo: any = null;

    attacker_priority = 0;
    defender_priority = 0;
	attackerSpeedValue = 0;
    attacker_head_pic = 0;
    defender_head_pic = 0;

}

File: src/client/battle/report/BattleCache.lua

BattleCache = class("BattleCache")

function BattleCache:ctor()
    print("创建BattleCache")
    --用户信息
    self.userInfo =  nil

    --攻击方名称
    self.attackName =  ""

    --被攻击方名称
    self.byAttackName =  ""

    --战斗类型
    self.battleType =  FightModule.FIGHT_TYPE_PVE_NORMAL

    --攻击方战力
    self.attackCombatForce =  0

    --被攻击方战力
    self.byAttackComobatForce =  0

    --被攻击者Id
    self.byAttackerId =  nil

    --当前战斗场次
    self.currentBattleCount =  0

    --最大战斗场次
    self.maxBattleCount =  0

    --难度
    self.difficulty =  1

    --阵型战斗对象
    self.attackerObjects =  {}

    --被攻击者战斗对象队列
    self.byAttackerObjectsList = {}

    --战斗中奖励卡牌
    self.battleReward = {}

    --战斗中奖励道具
    self.battleRewardProp = {}

    --战斗中奖励装备
    self.battleRewardEquipment = {}

    --总伤害
    self.totalDamage =  0

    self.userViceCaptainInfo =  nil

    self.byUserViceCaptainInfo =  nil

end

function BattleCache:addTotalDamage(totalDamage)
	self.totalDamage = self.totalDamage + totalDamage
end


function BattleCache:addCurrentBattleCount(currentBattleCount)
	self.currentBattleCount = self.currentBattleCount + currentBattleCount
end

function BattleCache:getFightObjectCount(fightObjects)
	local count = 0
	for k, v in pairs(fightObjects)  do
		if (nil ~= v) then
			count = count + 1
		end
	end
	return count
end

